geonode/layers/utils.py

```diff
@@ -653,51 +653,59 @@ def upload(incoming, user=None, overwrite=False,
     return output
 
 
-def create_thumbnail(instance, thumbnail_remote_url, thumbnail_create_url=None, check_bbox=True, ogc_client=None):
-    if not ogc_client:
-        ogc_client = http_client
-    BBOX_DIFFERENCE_THRESHOLD = 1e-5
-
-    if not thumbnail_create_url:
-        thumbnail_create_url = thumbnail_remote_url
-
-    if check_bbox:
-        # Check if the bbox is invalid
-        valid_x = (
-            float(
-                instance.bbox_x0) -
-            float(
-                instance.bbox_x1)) ** 2 > BBOX_DIFFERENCE_THRESHOLD
-        valid_y = (
-            float(
-                instance.bbox_y1) -
-            float(
-                instance.bbox_y0)) ** 2 > BBOX_DIFFERENCE_THRESHOLD
-    else:
-        valid_x = True
-        valid_y = True
-
-    image = None
-
-    if valid_x and valid_y:
-        Link.objects.get_or_create(resource=instance.get_self_resource(),
-                                   url=thumbnail_remote_url,
-                                   defaults=dict(
-                                       extension='png',
-                                       name="Remote Thumbnail",
-                                       mime='image/png',
-                                       link_type='image',
+def create_thumbnail(instance, thumbnail_remote_url, thumbnail_create_url=None,
+                     check_bbox=True, ogc_client=None, overwrite=False):
+    thumbnail_dir = os.path.join(settings.MEDIA_ROOT, 'thumbs')
+    thumbnail_name = 'layer-' + instance.uuid + '-thumb.png'
```

Comment:
Trivial nitpick: for readability & consistency, how about switching this from concatenating ("+") to substituting instance.uuid in with `%s`? That way it'll match how we form `filename` elsewhere in this same file (https://github.com/travislbrundage/geonode/blob/315cd1c7ab6a7a7de441ff43f1ea1410171d244e/geonode/layers/utils.py#L710)